python/ray/ray_operator/operator.py

```diff
@@ -181,162 +170,114 @@ def write_config(self) -> None:
             yaml.dump(self.config, file)
 
     def delete_config(self) -> None:
-        os.remove(self.config_path)
+        try:
+            os.remove(self.config_path)
+        except OSError:
+            log_prefix = ",".join([self.name, self.namespace])
+            logger.warning(f"{log_prefix}: config path does not exist {self.config_path}")
 
-    def set_generation(self, generation: int) -> None:
-        self._generation = generation
 
-    def set_num_retries(self, num_retries: int) -> None:
-        self._num_retries = num_retries
+# Maps ray cluster (name, namespace) pairs to RayCluster python objects.
+# TODO: decouple monitoring background thread into a kopf.daemon and move this into
+#  the memo state.
```

Comment:
Actually, can we move the RayCluster objects into memo state in this PR?

Specifically, I mean getting rid of the `ray_clusters` global variable and instead doing
`memo.ray_cluster = RayCluster(cluster_config)` to initialize one of these objects, and accessing `memo.ray_cluster`
in the update events. 

Per-resource memo state looks like the right idiom for what the `ray_clusters` dict was trying to do.

Reply:
Fixed.